src/Bedrock.Framework/Client/ClientBuilder.cs

```diff
@@ -18,7 +18,7 @@ public ClientBuilder(IServiceProvider serviceProvider)
 
         internal static object Key { get; } = new object();
 
-        private IConnectionFactory ConnectionFactory { get; set; } = new ThrowConnectionFactory();
+        internal IConnectionFactory ConnectionFactory { get; set; } = new ThrowConnectionFactory();
```

Comment:
Undo, there's a method for this already call for this. `UseConnectionFactory(factory => new MyFactory(factory));`